mzLib/SpectralAveraging/Algorithms/AveragedSpectraWriter.cs

```diff
@@ -14,82 +15,93 @@ public static class AveragedSpectraWriter
     /// <param name="averagedScans"></param>
     /// <param name="parameters"></param>
     /// <param name="originalSpectraPath"></param>
+    /// <param name="destinationDirectory"></param>
+    /// <param name="averagedFileName"></param>
     /// <exception cref="NotImplementedException"></exception>
     public static void WriteAveragedScans(MsDataScan[] averagedScans, SpectralAveragingParameters parameters,
-        string originalSpectraPath)
+        string originalSpectraPath, string? destinationDirectory = null, string? averagedFileName = null)
     {
         switch (parameters.OutputType)
         {
             case OutputType.MzML:
-                WriteAveragedSpectraAsMzML(averagedScans, originalSpectraPath);
-                break;
-
-            case OutputType.Text:
-                WriteAveragedSpectraAsTxtFile(averagedScans, parameters, originalSpectraPath);
+                WriteAveragedSpectraAsMzMl(averagedScans, originalSpectraPath, destinationDirectory, averagedFileName);
                 break;
 
             default: throw new MzLibException("Output averaged scans type not implemented");
         }
     }
 
     /// <summary>
-    ///     Outputs averaged spectra in mzml format to same location
+    /// Outputs averaged spectra in mzml format to same location as original spectra unless
+    /// specified by the optional destination directory and destination file name fields
     /// </summary>
     /// <param name="averagedScans"></param>
     /// <param name="originalSpectraPath"></param>
-    private static void WriteAveragedSpectraAsMzML(MsDataScan[] averagedScans,
-        string originalSpectraPath)
+    /// <param name="destinationDirectory"></param>
+    /// <param name="averagedFileName"></param>
+    private static void WriteAveragedSpectraAsMzMl(MsDataScan[] averagedScans,
+        string originalSpectraPath, string? destinationDirectory = null, string? averagedFileName = null)
```

Comment:
why leave filename and path as null?

Reply:
By default, those can be unspecified. This results in the averaged spectra being placed in the same location as your original spectra with the file name being "{original file name}-averaged".

This allows more flexibility for developers using the writer in the future. I can add that to the comments